src/components/Entities/EntityExecutions.tsx

```diff
@@ -50,6 +50,11 @@ export const EntityExecutions: React.FC<EntityExecutionsProps> = ({ id }) => {
         }
     );
 
+    /** Don't render component until finish fetching user profile */
+    if (filtersState.filters[4].status !== 'LOADED') {
```

Comment:
You could just use `useUserProfile` directly here and check the status on that. We don't need to add status to a filter and pipe it through.
All react-query queries for a particular piece of data are synchronized and will receive the same values.
```tsx
const { isLoading: userProfileIsLoading } = useUserProfile();
if (userProfileIsLoading) {
    return null;
}
```